code/modules/integrated_electronics/subtypes/lists.dm

```diff
@@ -275,19 +297,24 @@
 	var/number_of_pins = 16
 
 /obj/item/integrated_circuit/lists/constructor/Initialize()
-	for(var/i = 1 to number_of_pins)
-		inputs["input [i]"] = IC_PINTYPE_ANY // This is just a string since pins don't get built until ..() is called.
+	for(var/i in 1 to number_of_pins)
+		inputs["key [i]"] = IC_PINTYPE_ANY // This is just a string since pins don't get built until ..() is called.
+	for(var/i in 1 to number_of_pins)
+		inputs["value [i]"] = IC_PINTYPE_ANY
```

Comment:
are you sure, that it's neccessary?

Reply:
yes otherwise it can't construct assoc lists